plugins/kubernetes/src/alpha/apis.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createApiExtension,
+  createApiFactory,
+  discoveryApiRef,
+  fetchApiRef,
+} from '@backstage/frontend-plugin-api';
+import {
+  KubernetesBackendClient,
+  kubernetesApiRef,
+  kubernetesProxyApiRef,
+  kubernetesAuthProvidersApiRef,
+  KubernetesAuthProviders,
+  KubernetesProxyClient,
+  kubernetesClusterLinkFormatterApiRef,
+  getDefaultFormatters,
+  KubernetesClusterLinkFormatter,
+  DEFAULT_FORMATTER_NAME,
+} from '@backstage/plugin-kubernetes-react';
+import {
+  gitlabAuthApiRef,
+  googleAuthApiRef,
+  microsoftAuthApiRef,
+  oktaAuthApiRef,
+  oneloginAuthApiRef,
+} from '@backstage/core-plugin-api';
+
+export const kubernetesApiExtension = createApiExtension({
+  factory: createApiFactory({
+    api: kubernetesApiRef,
+    deps: {
+      discoveryApi: discoveryApiRef,
+      fetchApi: fetchApiRef,
+      kubernetesAuthProvidersApi: kubernetesAuthProvidersApiRef,
+    },
+    factory: ({ discoveryApi, fetchApi, kubernetesAuthProvidersApi }) =>
+      new KubernetesBackendClient({
+        discoveryApi,
+        fetchApi,
+        kubernetesAuthProvidersApi,
+      }),
+  }),
+});
+
+export const kubernetesProxyApi = createApiExtension({
+  factory: createApiFactory({
+    api: kubernetesProxyApiRef,
+    deps: {
+      kubernetesApi: kubernetesApiRef,
+    },
+    factory: ({ kubernetesApi }) =>
+      new KubernetesProxyClient({
+        kubernetesApi,
+      }),
+  }),
+});
+
+export const kubernetesAuthProvidersApi = createApiExtension({
+  factory: createApiFactory({
+    api: kubernetesAuthProvidersApiRef,
+    deps: {
+      gitlabAuthApi: gitlabAuthApiRef,
+      googleAuthApi: googleAuthApiRef,
+      microsoftAuthApi: microsoftAuthApiRef,
+      oktaAuthApi: oktaAuthApiRef,
+      oneloginAuthApi: oneloginAuthApiRef,
+    },
+    factory: ({
+      gitlabAuthApi,
+      googleAuthApi,
+      microsoftAuthApi,
+      oktaAuthApi,
+      oneloginAuthApi,
+    }) => {
+      const oidcProviders = {
```

Comment:
Wonder if it could be worth making this extensible in a simpler way, now that we have the capability to build that in the framework? @mclarke47 

A bit similar to how this works: https://github.com/backstage/community-plugins/blob/24c529dfdc61c994e24945217de81eb129176c86/workspaces/graphiql/plugins/graphiql/src/alpha.tsx#L80

Doesn't need to be part of this PR